scipy/sparse/csgraph/tests/test_matching.py

```diff
@@ -143,3 +144,54 @@ def test_large_random_graph_with_one_edge_incident_to_each_vertex():
     # Should get identity matrix back
     assert_equal(any(C1.diagonal() == 0), False)
     assert_equal(any(C2.diagonal() == 0), False)
+
+
+def test_min_weight_full_matching_empty_graph():
+    biadjacency_matrix = csr_matrix((0, 0))
+    row_ind, col_ind = min_weight_full_bipartite_matching(biadjacency_matrix)
+    assert len(row_ind) == 0
+    assert len(col_ind) == 0
+
+
+def test_min_weight_full_matching_empty_left_partition():
+    biadjacency_matrix = csr_matrix((2, 0))
+    row_ind, col_ind = min_weight_full_bipartite_matching(biadjacency_matrix)
+    assert len(row_ind) == 0
+    assert len(col_ind) == 0
+
+
+def test_min_weight_full_matching_empty_right_partition():
+    biadjacency_matrix = csr_matrix((0, 3))
+    row_ind, col_ind = min_weight_full_bipartite_matching(biadjacency_matrix)
+    assert len(row_ind) == 0
+    assert len(col_ind) == 0
+
+
+def test_min_weight_full_matching_infeasible_square():
+    biadjacency_matrix = csr_matrix([[1, 1, 1], [1, 0, 0], [1, 0, 0]])
+    with pytest.raises(ValueError):
+        min_weight_full_bipartite_matching(biadjacency_matrix)
+
+
+def test_min_weight_full_matching_infeasible_other_square():
+    biadjacency_matrix = csr_matrix([[1, 1, 1], [0, 0, 1], [0, 0, 1]])
+    with pytest.raises(ValueError):
+        min_weight_full_bipartite_matching(biadjacency_matrix)
+
+
+def test_min_weight_full_matching_infeasible_rectangular():
+    biadjacency_matrix = csr_matrix([[1, 0, 0], [2, 0, 0]])
+    with pytest.raises(ValueError):
+        min_weight_full_bipartite_matching(biadjacency_matrix)
+
+
+def test_min_weight_full_matching_infeasible_other_rectangular():
+    biadjacency_matrix = csr_matrix([[1, 0], [2, 0], [5, 0]])
+    with pytest.raises(ValueError):
+        min_weight_full_bipartite_matching(biadjacency_matrix)
```

Comment:
same for the 4 functions above I suppose, since they all raise ValueError with same function with only the csr_matrix changing, but minor point..

Reply:
Changed to
```python
@pytest.mark.parametrize('biadjacency_matrix',
                         [
                            [[1, 1, 1], [1, 0, 0], [1, 0, 0]],
                            [[1, 1, 1], [0, 0, 1], [0, 0, 1]],
                            [[1, 0, 0], [2, 0, 0]],
                            [[1, 0], [2, 0], [5, 0]]
                         ])
def test_min_weight_full_matching_infeasible_problems(biadjacency_matrix):
    with pytest.raises(ValueError):
        min_weight_full_bipartite_matching(csr_matrix(biadjacency_matrix))
```